src/vfs2.c

```diff
@@ -1727,6 +1727,7 @@ int vfs2_add_uncommitted(sqlite3_file *file,
 	 * (wal_cursor). */
 	e->shm_locks[WAL_WRITE_LOCK] = VFS2_EXCLUSIVE;
 
+	uint32_t start = e->wal_cursor;
 	struct wal_index_full_hdr *ihdr = get_full_hdr(e);
 	uint32_t mx = ihdr->basic[0].mxFrame;
 	if (mx > 0 && ihdr->nBackfill == mx) {
```

Comment:
[Nit]: Should we add `start != 0 && ...` for clarity with the condition below? That way we don't have to remember that mx < wal_cursor && mx > 0 => wal_cursor > 0

Reply:
Yes, thanks, this could use refactoring.

Reply:
This logic has been substantially re-organized in a follow-up commit, hopefully it's more clear now!